source/codegen/metadata/nidaqmx/functions.py

```diff
@@ -125,6 +125,56 @@
         ],
         'returns': 'int32'
     },
+    'ReadAnalogF64': {
+        'parameters': [
+            {
+                'direction': 'in',
+                'name': 'task',
+                'type': 'TaskHandle'
+            },
+            {
+                'direction': 'in',
+                'name': 'numSampsPerChan',
+                'type': 'int32'
+            },
+            {
+                'direction': 'in',
+                'name': 'timeout',
+                'type': 'float64'
+            },
+            {
+                'direction': 'in',
+                'enum': 'GroupBy',
+                'name': 'fillMode',
+                'type': 'int32'
+            },
+            {
+                'direction': 'out',
+                'name': 'readArray',
+                'size': {
+                    'mechanism': 'passed-in',
+                    'value': 'arraySizeInSamps'
+                },
+                'type': 'float64[]'
+            },
+            {
+                'direction': 'in',
+                'name': 'arraySizeInSamps',
+                'type': 'uInt32'
+            },
+            {
+                'direction': 'out',
+                'name': 'sampsPerChanRead',
+                'type': 'int32'
+            },
+            {
+                'direction': 'pass_null',
```

Comment:
Note: this is documented in the [C API](https://zone.ni.com/reference/en-XX/help/370471AM-01/daqmxcfunc/daqmxreadanalogf64/). The desired behavior for grpc-device is to omit the parameter from the proto and hardcode a `nullptr` input.

I used `direction` for this for a few reasons. (1) the actual "direction" on the API is ambiguous, especially since ptrs are typically outputs. (2) it codegens easier this way.

Reply:
> it codegens easier this way

It would be *almost* as easy to implement it as an `in` parameter with `pass_null: True`. There'd be a few more places where it needs to be removed from parameter lists (i.e., with `include_in_proto: False`)

Reply:
Yeah, in the C API linked it's under the _Input_ parameters so I'm leaning towards keeping it that way. I think I'd prefer that as well because the direction being "pass_null" just sounds a little awkward to me. And I agree, `pass_null: True` sounds good for the indicator. I guess we could be more general with a `value: nullptr` but the pattern we probably'd rather capture is the pass_null part like you suggest.

Reply:
Good call, thanks for the feedback.

Addressed in [this change](https://github.com/ni/grpc-device/pull/251/commits/21ac3765c6e9023115cd41ed27b7ab7558df3539)

I like the `value: nullptr` idea too. The advantage of `pass_null` is that it captures both the `input_by_ptr` and `hardcoded_value` concept in one field. Of course we could also handle that with 2 fields, but I'd question whether that really generalizes to anything useful (at least for `input_by_ptr`; `hardcoded_value` seems pretty reusable).